compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/types/BLangFunctionTypeNode.java

```diff
@@ -37,10 +39,13 @@
  */
 public class BLangFunctionTypeNode extends BLangType implements FunctionTypeNode {
 
-    public List<BLangVariable> params = new ArrayList<>();
+    public List<BLangSimpleVariable> params = new ArrayList<>();
     public BLangVariable restParam;
     public BLangType returnTypeNode;
     public Set<Flag> flagSet = new HashSet<>();
+    public BInvokableTypeSymbol symbol;
```

Comment:
Why is the type symbol being added to the type node? Why can't we get the type and then get the type symbol?

Reply:
We need to create a separate env for `functiontypenode`. For that, I use this field. Yes, We can get it from the type. I will remove this.